### PR TITLE
Update README.md for Metricbeat on K8s

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Updated README.md file for Metricbeat on Kubernetes to explain to the user how to run Metricbeat locally or on cloud. 

## Why is it important?

The updated file makes clear to the user how to run Metricbeat on Kubernetes.

## Checklist

- [x] ~~My code follows the style guidelines of this project~~
- [x] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [x] ~~I have made corresponding change to the default configuration files~~
- [x] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Test example on kind and on GKE

### CODE REVIEW

File: deploy/kubernetes/metricbeat/README.md [hunk 1]
```diff
@@ -2,30 +2,145 @@
...
+  This files uses a set of environment variables to configure Elasticsearch output.
 
-Everything is deployed under `kube-system` namespace, you can change that by
-updating YAML manifests under this folder.
+| Variable               | Default       | Description                          |
+|------------------------|---------------|--------------------------------------|
+| ELASTICSEARCH_HOST     | elasticsearch | Elasticsearch host                   |
+| ELASTICSEARCH_PORT     | 9200          | Elasticsearch port                   |
+| ELASTICSEARCH_USERNAME | elastic       | Elasticsearch username for HTTP auth |
+| ELASTICSEARCH_PASSWORD | changeme      | Elasticsearch password               |
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:17]: could you please explain that these are two sets of variables either host, port, username and password for managed installation or cloud_id and cloud_auth for cloud installation. 
It would be great also to add a link to both sets of configs either [standalone](https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html) or [managed/cloud](https://www.elastic.co/guide/en/beats/metricbeat/current/configure-cloud-id.html)

File: deploy/kubernetes/metricbeat/README.md [hunk 2]
```diff
@@ -2,30 +2,145 @@
...
+- role: control-plane
+- role: worker
+- role: worker
+- role: worker
+```
+
+After that, we can create a cluster using the command below.
+
+```
+$ kind create cluster --config=cluster-config.yaml
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:51]: I usually tend to fix the k8s cluster by passing a specific version with `--image=kindest/node:v1.23.5`

File: deploy/kubernetes/metricbeat/README.md [hunk 3]
```diff
@@ -2,30 +2,145 @@
...
+
+#### Run Metricbeat on Kubernetes
+
+You can check the nodes of the newly created cluster by running the following:
+
+```
+$ kubectl get nodes
+```
+
+Metricbeat gets some metrics from [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics). You can download the project. You can run, for example:
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:79]: Can you please explain this concept a bit better?
I would suggest that we rephrase this to be...

`you can download the project` => `git clone the repo locally and then run kubectl apply....`

File: deploy/kubernetes/metricbeat/README.md [hunk 4]
```diff
@@ -2,30 +2,145 @@
...
+
+#### Visualizing data in Kibana
+
+To visualize your data in Kibana, we will use Elastic Cloud. For that do the following:
+1. [Log in](https://cloud.elastic.co/home) to your Elastic Cloud account.
+2. Select the target deployment.
+3. Navigate to the **Analytics** endpoint and select **Discover**. To see Metricbeat data, make sure the predefined `metricbeat-*` index pattern is selected.
+4. To load a Kibana dashboard, you have to select **Dashboard** on the side menu. For example, you can see data about your Metricbeat DaemonSet on the dashboard *[Metricbeat Kubernetes] DaemonSets*. If, by any chance, you don't have dashboards downloaded in Kibana, you can go [here](https://www.elastic.co/guide/en/beats/metricbeat/current/load-kibana-dashboards.html) to find a solution for that.
+
+If you can't see any data on the index, then you probably haven't set the YAML manifest file as you should. To do that, open the [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml) and update the following:
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:115]: Personally I would rephrase this bit.
If you are using Elastic cloud, you need first to deploy your Elastic cloud cluster to get the Id and username/password before you can deploy the elastic agent, you can't do the other way around.

So I suggest that instead of saying, "if you can't see any data ...", (since there is no other way that could work)... `Now deploy the elastic agent with these credentials from the newly created Elastic cloud...`

**Review by gsantoro — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/metricbeat/README.md [hunk 5]
```diff
@@ -2,30 +2,145 @@
...
+$ kubectl apply -f beats/deploy/kubernetes/metricbeat/metricbeat-daemonset.yaml
+```
+
+You should now be able to visualize your data on Kibana.
 
-We use official [Beats Docker images](https://github.com/elastic/beats-docker),
-as they allow external files configuration, a [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/)
-is used for kubernetes specific settings. Check [metricbeat-daemonset-configmap.yaml](metricbeat-daemonset-configmap.yaml)
-for details.
+#### Kubernetes dashboard
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:131]: I find this bit confusing here since this refers to the Out of the Box Kubernetes dashboard and we have a Kubernetes dashboard as part of our K8s integration. On top of that, I am not sure that deploying other manifests is a good idea in a readme like this. Personally I would have added just a link to either https://kubernetes.io/docs/tasks/access-application-cluster/web-ui-dashboard/ or even better suggest to use https://k9scli.io/ or https://k8slens.dev/ in order to avoid adding more resources to the current cluster

File: deploy/kubernetes/metricbeat/README.md [hunk 6]
```diff
@@ -2,30 +2,215 @@
...
 
-By deploying metricbeat as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/)
-we ensure we get a running metricbeat daemon on each node of the cluster.
 
-Everything is deployed under `kube-system` namespace, you can change that by
-updating YAML manifests under this folder.
+We use official [Metricbeat Docker images](https://www.docker.elastic.co/r/beats/metricbeat), as they allow external files' configuration. Our YAML manifests are the following:
 
-### Settings
+<ul>
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:10]: I don't think I ever seen embedded html in our markdown readmes. do you mind removing it and replacing with markdown syntax? thanks

Comment (constanca-m) [reply deploy/kubernetes/metricbeat/README.md:10]: Of course I can change that, but then I don't think I have a way to have multiple lines in one point of the list. I tried looking for a solution, but couldn't find any

File: deploy/kubernetes/metricbeat/README.md [hunk 7]
```diff
@@ -7,49 +7,29 @@
...
+  * [Self-managed](https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html) Elasticsearch service:
 
-<ul>
-<li>
-
-[Self-managed](https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html) Elasticsearch service:
+    | Variable               | Default       | Description                          |
+    |------------------------|---------------|--------------------------------------|
+    | ELASTICSEARCH_HOST     | elasticsearch | Elasticsearch host                   |
+     | ELASTICSEARCH_PORT     | 9200          | Elasticsearch port                   |
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:19]: can you realign this table, please?

File: deploy/kubernetes/metricbeat/README.md [hunk 8]
```diff
@@ -161,53 +141,41 @@ $ kubectl get pods -o wide --namespace=kube-system | grep ^metricbeat
...
-<li>
+1. [Log in](https://cloud.elastic.co/home) to your Elastic Cloud account.
 
-On the deployment overview page, copy down the Cloud ID.
-</li>
-<li>
+2. Create a [deployment](https://www.elastic.co/guide/en/cloud/current/ec-create-deployment.html). Make sure to save the credentials.
 
-Set the environment variables on the [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml) file:
+3. On the deployment overview page, copy down the Cloud ID.
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:150]: since you mentioned to copy the cloud Id, maybe we should mention as well where to get the clout Auth variables from (eg. during the deployment process, it provides you those variables or it allows you to download a file locally.

File: deploy/kubernetes/metricbeat/README.md [hunk 9]
```diff
@@ -2,30 +2,183 @@
...
+* [metricbeat-daemonset-configmap.yaml](metricbeat-daemonset-configmap.yaml) to create the [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) necessary to [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml).
+
+* [metricbeat-daemonset.yaml](metricbeat-daemonset.yaml) to create a [DeamonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/). This way we ensure we get a running metricbeat daemon on each node of the cluster.
+This file uses a set of environment variables to configure Elasticsearch output. We have two different sets:
+  * [Self-managed](https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html) Elasticsearch service:
+
+    | Variable               | Default       | Description                          |
+    |------------------------|---------------|--------------------------------------|
+    | ELASTICSEARCH_HOST     | elasticsearch | Elasticsearch host                   |
+     | ELASTICSEARCH_PORT     | 9200          | Elasticsearch port                   |
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:19]:
```suggestion
    | ELASTICSEARCH_PORT     | 9200          | Elasticsearch port                   |
```

File: deploy/kubernetes/metricbeat/README.md [hunk 10]
```diff
@@ -2,30 +2,183 @@
...
+     | ELASTICSEARCH_USERNAME | elastic       | Elasticsearch username for HTTP auth |
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:20]:
```suggestion
    | ELASTICSEARCH_USERNAME | elastic       | Elasticsearch username for HTTP auth |
```

File: deploy/kubernetes/metricbeat/README.md [hunk 11]
```diff
@@ -2,30 +2,183 @@
...
+     | ELASTICSEARCH_PASSWORD | changeme      | Elasticsearch password               |
```
Comment (gsantoro) [inline deploy/kubernetes/metricbeat/README.md:21]:
```suggestion
    | ELASTICSEARCH_PASSWORD | changeme      | Elasticsearch password               |
```

**Review by gsantoro — APPROVED** [review]
